test/grpc/integration/stub_test.exs

```diff
@@ -78,7 +80,7 @@ defmodule GRPC.Integration.StubTest do
 
   test "returns error when timeout" do
     run_server(SlowServer, fn port ->
-      {:ok, channel} = GRPC.Stub.connect("localhost:#{port}")
+      {:ok, channel} = GRPC.Stub.connect("http://localhost:#{port}")
```

Comment:
Why did we have to change these to http?

Reply:
I was just updating it to avoid using the deprecated branch

Reply:
Not sure if we should really deprecate this. Any specific reason?

Reply:
I'm not 100% sure if it should be deprecated. I'm aiming for an API where the input address is always a valid URI. For example, this would be the code without backwards compatibility:

```elixir
case URI.parse(addr) do
   %URI{scheme: "http"} -> # insecure 
   %URI{scheme: "https"} -> # secure
   %URI{scheme: nil, path: path} -> # unix domain socket
   _ -> # error
end
```

Reply:
If you think deprecation isn't the way, I can revert ec02a949c173ca17f563e1504aa5078a700af94a and remove the warning log

Reply:
I see no need to deprecate since:

1. It's an unnecessary breaking change because your original issue was not being able to pass a scheme
2. The code to maintain this is quite small.

The third alternative where you only have the path but no port is already captured in the fallback clause.

Reply:
Let's revert for now